git-ftp

```diff
@@ -811,6 +819,40 @@ handle_remote_protocol_options() {
 	[ "$CURL_INSECURE" != "0" ] && REMOTE_CMD_OPTIONS="$REMOTE_CMD_OPTIONS -k"
 }
 
+extract_folder_snapshot() {
+    local fulladdress=$1
+    SNAPSHOT_DIR="$(basename "$fulladdress")"
+}
+
+save_credential() {
+    git config git-ftp.user $REMOTE_USER
+    git config git-ftp.url $URL
+    git config git-ftp.password $REMOTE_PASSWD
+}
+
+init_new_repository() {
+    
+    [ -z "$URL" ] && print_error_and_die "Error: give a url to snapshot." "$ERROR_USAGE"
+    [ -z $SNAPSHOT_DIR ] && extract_folder_snapshot $URL   # empty string, "clone" into folder with the same name as in the server
+    
+    if [ -d "$SNAPSHOT_DIR" ]; then    # if directory already exists (including . ) check for emptiness
```

Comment:
Another way to ensure that a directory exists:
```bash
mkdir -p "$SNAPSHOT_DIR" || print_error_and_die "..."
```
That also works to create multiple directories. For example if `/tmp` exists and someone chooses to download into `/tmp/experiments/git-projects/new/html`.